packages/web3/src/address/address.test.ts

```diff
@@ -77,134 +150,237 @@ describe('address', function () {
+    })
+  })
+
+  describe('groupOfLockupScript', () => {
+    it('should calculate the group of lockup script', () => {
+      const bytes0 = new Uint8Array(randomBytes(32))
+      const bytes1 = new Uint8Array(randomBytes(32))
+      const bytes2 = new Uint8Array(randomBytes(32))
+
+      const p2pkh: LockupScript = { kind: 'P2PKH', value: new Uint8Array(bytes0) }
+      const p2pkhAddress = bs58.encode(new Uint8Array([0x00, ...bytes0]))
+      expect(groupOfAddress(p2pkhAddress)).toBe(groupOfLockupScript(p2pkh))
+
+      const p2mpkh: LockupScript = {
+        kind: 'P2MPKH',
+        value: { publicKeyHashes: [bytes0, bytes1, bytes2], m: 2 }
+      }
+      const p2mpkhAddress = bs58.encode(new Uint8Array([0x01, 0x03, ...bytes0, ...bytes1, ...bytes2, 0x02]))
+      expect(groupOfAddress(p2mpkhAddress)).toBe(groupOfLockupScript(p2mpkh))
+
+      const p2sh: LockupScript = { kind: 'P2SH', value: bytes0 }
+      const p2shAddress = bs58.encode(new Uint8Array([0x02, ...bytes0]))
+      expect(groupOfAddress(p2shAddress)).toBe(groupOfLockupScript(p2sh))
+
+      const p2c: LockupScript = { kind: 'P2C', value: bytes0 }
+      const p2cAddress = bs58.encode(new Uint8Array([0x03, ...bytes0]))
+      expect(groupOfAddress(p2cAddress)).toBe(groupOfLockupScript(p2c))
+    })
+  })
+
+  describe('groupOfPrivateKey', () => {
+    it('should calculate the group of private key', () => {
+      expect(groupOfPrivateKey('91411e404289ec0e8b3058697f53f9b26fa7305158b4ef1a81adfbabcf094e49')).toEqual(0)
+      expect(groupOfPrivateKey('8b63d2f60d74d9d74b9132bd1e4d1656d4ab1e8c2efc1a56e7d9142bf2fa2c12')).toEqual(1)
+      expect(groupOfPrivateKey('82e1f50a8e372933ab8afc6362934b693e5aaa4ca308b0aac27fbe8755e0a3fa')).toEqual(2)
+      expect(groupOfPrivateKey('a5c91afebe25e1644e9023e0d341ca713b59beca36e6635de37ef0c9c8689654')).toEqual(3)
+    })
+  })
+
+  describe('publicKeyFromPrivateKey', () => {
+    it('should compute public key from private key', () => {
+      expect(publicKeyFromPrivateKey('91411e484289ec7e8b3058697f53f9b26fa7305158b4ef1a81adfbabcf090e45')).toBe(
+        '030f9f042a9410969f1886f85fa20f6e43176ae23fc5e64db15b3767c84c5db2dc'
+      )
+    })
+  })
+
+  describe('publicKeyFromPrivateKey', () => {
+    it('should compute public key from private key', () => {
+      expect(publicKeyFromPrivateKey('91411e484289ec7e8b3058697f53f9b26fa7305158b4ef1a81adfbabcf090e45')).toBe(
+        '030f9f042a9410969f1886f85fa20f6e43176ae23fc5e64db15b3767c84c5db2dc'
+      )
+    })
+  })
+  describe('addressFromPublicKey', () => {
+    it('should compute address from public key', () => {
+      expect(addressFromPublicKey('030f9f042a9410969f1886f85fa20f6e43176ae23fc5e64db15b3767c84c5db2dc')).toBe(
+        '1ACCkgFfmTif46T3qK12znuWjb5Bk9jXpqaeWt2DXx8oc'
+      )
+    })
+
+    it('should throw error for invalid public key', () => {
+      expect(() => tokenIdFromAddress('eBrjfQNeyUCuxE4zpbfMZcbS3PuvbMJDQBCyk4HRHtX4')).toThrow(
```

Comment:
Shall we test `addressFromPublicKey` function here?

Reply:
removing this test, because the `addressFromPublicKey` does not have error catching logic, so all string will be accepted as valid pubkeys